fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/BaseCompositeAvailability.java

```diff
@@ -0,0 +1,70 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.availability;
+
+import com.yahoo.bard.webservice.data.config.names.TableName;
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList;
+import com.yahoo.bard.webservice.util.StreamUtils;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A class implementing common capabilities for availabilities backed by a collection of other availabilities.
+ */
+public abstract class BaseCompositeAvailability implements Availability {
+
+    private final Set<Availability> sourceAvailabilities;
+    private final Set<TableName> dataSourcesNames;
+
+    /**
+     * Constructor.
+     *
+     * @param availabilityStream  A potentially ordered stream of availabilities which supply this composite view
+     */
+    protected BaseCompositeAvailability(Stream<Availability> availabilityStream) {
```

Comment:
The general principal is to have interfaces be only as restrictive as required. If all this thing needs is, essentially, an iterator, and even then, if order doesn't really matter, a stream is a perfect fit here. Essentially, `Stream` is the better / more event-based version of `Collection`.